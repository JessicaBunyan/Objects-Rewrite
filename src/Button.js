import React, { Component } from "react";

export class Button extends Component {
  render() {
    return (
      <div className="button" onClick={this.props.onClick}>
        {this.props.children}
      </div>
    );
  }
}
