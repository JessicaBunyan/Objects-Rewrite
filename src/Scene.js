import React, { Component } from "react";

export class Scene extends Component {
  constructor(props) {
    super(props);
  }
  render() {
    return (
      <div
        className={"scene " + (this.props.active == true ? "active" : "")}
        onClick={() => this.props.onClick()}
      >
        {this.props.children}
      </div>
    );
  }
}
