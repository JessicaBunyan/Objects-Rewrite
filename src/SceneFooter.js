import React, { Component } from "react";
import backChevron from "./img/back-chevron.png";

const SceneFooter = ({ action }) => (
  <div className="footer" onClick={() => action()}>
    <img className="back-chevron" src={backChevron} />
  </div>
);

export default SceneFooter;
