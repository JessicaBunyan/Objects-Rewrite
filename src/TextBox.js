import React, { Component } from "react";
import { calcColour } from "./utils";
import colourDrop from "./img/colourDrop.png";

export class TextBox extends Component {
  render() {
    return <div className={"text-box"}>{this.props.children}</div>;
  }
}
