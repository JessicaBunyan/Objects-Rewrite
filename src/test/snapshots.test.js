import React from "react";
import renderer from "react-test-renderer";
import { TextBox } from "../TextBox";
import { Scene } from "../Scene";
import SceneFooter from "./../SceneFooter";
import { Button } from "./../Button";

it("TextBox renders correctly", () => {
  const tree = renderer
    .create(
      <TextBox className="d1">
        <h2>Test</h2>
        <h3>textbox</h3>
      </TextBox>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("SceneFooter renders correctly", () => {
  const tree = renderer
    .create(
      <SceneFooter
        onClick={() => {
          return;
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("Button no activity renders correctly", () => {
  const tree = renderer
    .create(
      <Button
        onClick={() => {
          return;
        }}
      >
        <h2>text</h2>
      </Button>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("Button pressed renders correctly", () => {
  const tree = renderer
    .create(
      <Button
        onClick={() => {
          return;
        }}
        pressed={true}
      >
        <h2>text</h2>
      </Button>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("Button failpressed renders correctly", () => {
  const tree = renderer
    .create(
      <Button
        onClick={() => {
          return;
        }}
        failPressed={true}
      >
        <h2>text</h2>
      </Button>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
