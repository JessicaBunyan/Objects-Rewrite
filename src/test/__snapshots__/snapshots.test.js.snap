// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button failpressed renders correctly 1`] = `
<div
  className=" button  fail-pressed"
  onClick={[Function]}
>
  <h2>
    text
  </h2>
</div>
------------HTML PREVIEW---------------
<div class=" button  fail-pressed">
  <h2>
    text
  </h2>
</div>
`;

exports[`Button no activity renders correctly 1`] = `
<div
  className=" button "
  onClick={[Function]}
>
  <h2>
    text
  </h2>
</div>
------------HTML PREVIEW---------------
<div class=" button ">
  <h2>
    text
  </h2>
</div>
`;

exports[`Button pressed renders correctly 1`] = `
<div
  className=" button  pressed "
  onClick={[Function]}
>
  <h2>
    text
  </h2>
</div>
------------HTML PREVIEW---------------
<div class=" button  pressed ">
  <h2>
    text
  </h2>
</div>
`;

exports[`SceneFooter renders correctly 1`] = `
<div
  className="footer"
  onClick={[Function]}
>
  <img
    className="back-chevron"
    src="back-chevron.png"
  />
</div>
------------HTML PREVIEW---------------
<div class="footer">
  <img
    class="back-chevron"
    src="back-chevron.png"
  >
</div>
`;

exports[`TextBox renders correctly 1`] = `
<div
  className=" text-box d1"
>
  <h2>
    Test
  </h2>
  <h3>
    textbox
  </h3>
</div>
------------HTML PREVIEW---------------
<div class=" text-box d1">
  <h2>
    Test
  </h2>
  <h3>
    textbox
  </h3>
</div>
`;
